docs/runbooks/VirtualMachineStuckInUnhealthyState.md

```diff
@@ -0,0 +1,251 @@
+is trying to provision resources, pull images, or schedule the workload.
+
+**Affected States:**
+- `Provisioning` - Resources (DataVolumes, PVCs) are being prepared
+- `Starting` - VM is attempting to start but no VMI exists yet
+- `Terminating` - VM is being deleted but without an active VMI
+- `Error` states - Various scheduling, image, or resource allocation errors
+
+## Impact
+
+- **Severity:** Warning
+- **User Impact:** VMs cannot start or are stuck in error states
+- **Business Impact:** Workloads cannot be deployed, affecting application
+  availability
+
+## Possible Causes
+
+### Resource-Related Issues
+- **Insufficient cluster resources** (CPU, memory, storage)
+- **Missing or misconfigured storage classes**
+- **PVC provisioning failures**
+- **DataVolume creation/import failures**
+
+### Image and Registry Issues  
+- **Container image pull failures** for containerDisk volumes
+- **Registry authentication problems**
+- **Network connectivity issues to image registries**
+- **Missing or corrupted VM disk images**
+
+### Scheduling and Node Issues
+- **No schedulable nodes available** (all nodes cordoned/unschedulable)
+- **Insufficient resources** like KVM/GPU on available nodes or a
+  mismatch between requested and available CPU models
+- **Node selector constraints** cannot be satisfied
+- **Taints and tolerations** preventing scheduling
+
+### Configuration Issues
+- **Invalid VM specifications** (malformed YAML, unsupported
+  features)
+- **Missing required Secrets or ConfigMaps**
+- **Incorrect resource requests/limits**
+- **Network configuration errors**
+
+## Diagnosis
+
+### 1. Check VM Status and Events
+```bash
+# Get VM details and status
+kubectl get vm <vm-name> -n <namespace> -o yaml
+
+# Check VM events for error messages
+kubectl describe vm <vm-name> -n <namespace>
+
+# Look for related events in the namespace
+kubectl get events -n <namespace> --sort-by='.lastTimestamp'
+```
+
+### 2. Verify Resource Availability
+```bash
+# Check node resources and schedulability
+kubectl get nodes -o wide
+kubectl describe nodes
+
+# Check storage classes and provisioners
+kubectl get storageclass
+kubectl get pv,pvc -n <namespace>
+
+# For DataVolumes (if using)
+kubectl get datavolume -n <namespace>
+kubectl describe datavolume <dv-name> -n <namespace>
+```
+
+### 3. Check Image Availability (for containerDisk)
+```bash
+# If using containerDisk, test image accessibility
+docker pull <vm-disk-image>
+
+# Check image pull secrets if required
+kubectl get secrets -n <namespace>
+```
+
+### 4. Verify KubeVirt Configuration
+```bash
+# Check KubeVirt installation status
+kubectl get kubevirt -n kubevirt
+
+# Verify virt-controller is running
+kubectl get pods -n kubevirt | grep virt-controller
+
+# Check virt-controller logs for errors
+kubectl logs -n kubevirt deployment/virt-controller
+
+
+# Verify virt-handler is running
+kubectl get pods -n kubevirt | grep virt-handler
```

Comment:
General comment: the kubevirt namespace is not always the installation namespace, and I think we should not guide user to use it. Other runbooks instruct how to get the Installation namespace, and we need to do the same here. 

This comment is for every usage of -n kubevirt in this runbook.